### PR TITLE
fix: include generic Newznab results for titles without a ruleset

### PR DESCRIPTION
**Pull request by fcandi**
## Description

A string/text search via the Newznab endpoint (`t=search` / `t=tvsearch`, as issued by Sonarr/Radarr when adding a series or movie) currently returns results **only** for titles that have a matching ruleset. For every other title the endpoint returns an empty RSS feed — even when the Mediathek API has plenty of hits. As a result, most public-broadcaster content cannot be found or grabbed through search in Sonarr/Radarr.

`fetchSearchResultsByString` now also emits API results that match no ruleset as **generic** Newznab items (1080p/720p/480p variants, mirroring the matched path). Common German episode patterns are parsed from the title into `season`/`episode` attributes:

- `(S01/E05)`, `S2026/E02`
- `Staffel 2 Folge 3`
- `Folge 5` / `Folge 5: <title>`
- `(4/6)` (episode/total)

A small related fix in `applyRulesetFilters`: a matched item is now removed from the unmatched list, so it can no longer appear twice (once matched, once generic).

## Type of change

- [x] Bug fix (no breaking changes; matched results are unchanged)

## Before / after

Live example against the real Mediathek API, query `Markus Lanz` (no ruleset), `t=tvsearch`:

**Before** (v1.2.1):

```xml
<newznab:response offset="0" total="0"/>
```

**After** (this PR) — `total="906"`, e.g.:

```xml
<item>
  <title>Markus.Lanz.S2026E70.Markus.Lanz.vom.11.Juni.2026.GERMAN.1080p.WEB.h264-MEDiATHEK</title>
  <guid isPermaLink="true">https://www.zdf.de/video/talk/markus-lanz-114/markus-lanz-vom-11-juni-2026-100#1080p</guid>
  <category>TV &gt; HD</category>
  <newznab:attr name="category" value="5000"/>
  <newznab:attr name="category" value="5040"/>
  <newznab:attr name="season" value="2026"/>
  <newznab:attr name="episode" value="70"/>
</item>
```

Same effect for `maybrit illner` (0 → 210) and `Panorama` (0 → 1840).

## Checklist

- [x] I have read CONTRIBUTING.md
- [x] `npm run lint` (0 errors)
- [x] `npm run format:check`
- [x] `npm run typecheck`
- [x] `npm run test` (incl. new tests for `generateGenericRssItems`)
- [x] `npm run build`
- [x] Tested locally (running in production since April 2026)

## Summary by CodeRabbit

* **New Features**
  * Unmatched search results can now generate generic RSS items for feeds.
  * Automatic season/episode extraction from various title patterns.
  * Multiple quality variants (1080p/720p/480p) with selectable preference.

* **Bug Fixes**
  * Matched items are no longer duplicated as generic/unmatched results.
  * Season-only or empty-text queries continue to return matched-only results (no generics).

### CODE REVIEW

File: src/services/newznab.ts [hunk 1]
```diff
@@ -500,6 +500,233 @@ export function generateMovieRssItems(
...
+  // Pattern 1: (S01/E05) or S01/E05
+  const sPattern = title.match(/\(?S(\d+)\/E(\d+)\)?/i);
+  if (sPattern) {
+    season = parseInt(sPattern[1], 10);
+    episode = parseInt(sPattern[2], 10);
+    episodeName = title.replace(sPattern[0], "").trim();
+  }
+
+  // Pattern 2: Staffel N Folge N
+  if (!episode) {
```
[the remark this answers is not in the document]
Comment (fcandi) [reply src/services/newznab.ts]: Good catch — fixed in 43762ec: the pattern cascade now guards on `episode === null` instead of a falsy check, so a correctly parsed episode 0 (special episodes) is no longer overwritten by a later pattern. Added a regression test (`"Folge 0 (2/6)"` → episode `00`) plus the suggested quality-variant and `Staffel N Folge N` edge cases.

**Review by SeoFood — CHANGES_REQUESTED** [review]
Requesting changes for one behavior regression found in live testing.

The intended generic search case works: I verified the PR build in Docker on `localhost:6768`, and `t=tvsearch&q=Markus%20Lanz&limit=5` returns `total="906"` with usable generic Newznab items.

However, season-only TV search now also emits generic unmatched results. For example, `GET /api/newznab?t=tvsearch&season=01&limit=5` returned `total="2778"` across unrelated shows. Since `caps` advertises `season` as a supported `tv-search` parameter, this can produce very noisy false positives for clients that issue season-only queries.

Please gate generic unmatched result generation behind a non-empty `q`, or otherwise preserve the previous behavior for season-only searches, and add a regression test for `t=tvsearch&season=01` without `q`.

Non-blocking: titles like `Staffel 2 Folge 3` currently become `S02E03.Staffel.2.Folge.3...`; it would be cleaner to omit the fallback episode-name segment when the pattern consumes the whole title.

**Comment by fcandi** [comment]
Thanks for the careful live test, you're right, that was a real regression.

Fixed in the latest commit: generic (no-ruleset) results are now only emitted when a non-empty `q` is present. A season-only query falls back to the previous matched-only behavior, so it no longer floods the feed.

The reason gating on `q` is sufficient: MediathekViewWeb AND-combines the entries in `queries[]`, so when both `q` and `season` are present the season acts as a filter that narrows results to the show rather than widening them. Quick check against the live API:

| queries | totalResults |
|---|---|
| `Tatort` (topic,title) | 1837 |
| `S01` (title) | 10000 (capped) |
| `Tatort` + `S01` | 55 |

Re-ran your exact tests on the PR build:

| query | before | after |
|---|---|---|
| `t=tvsearch&q=Markus%20Lanz&limit=5` | total="906" | total="909" (live drift, one new episode) |
| `t=tvsearch&season=01&limit=5` | total="2778" | **total="0"** |

Also added a regression test for `t=tvsearch&season=01` without `q` (plus an empty/whitespace-`q` case and a text-query happy-path) in `src/services/mediathek.test.ts`.

I also took the non-blocking nit: titles like `Staffel 2 Folge 3` no longer become `S02E03.Staffel.2.Folge.3`, the episode-name segment is omitted when the pattern consumes the whole title.

Full checklist green: lint, format:check, typecheck, test (incl. new ones), build.

File: src/services/mediathek.ts [hunk 2]
```diff
@@ -829,11 +832,22 @@ export async function fetchSearchResultsByString(
...
-  const response = convertItemsToRss(newznabItems, limit, offset);
+
+  // Generic (no-ruleset) results are only meaningful for an actual text search.
+  // A season-only query (e.g. tvsearch&season=01 with no q) would otherwise emit
+  // every title that merely contains "S01" across unrelated shows. Gate on a
+  // non-empty q to keep the previous (matched-only) behavior for those queries.
+  const hasTextQuery = !!q?.trim();
+  const genericItems: NewznabItem[] = hasTextQuery
+    ? unmatchedItems.flatMap((item) => generateGenericRssItems(item, quality))
+    : [];
```
[the remark this answers is not in the document]
Comment (fcandi) [reply src/services/mediathek.ts:850]: @/tmp/cr-reply.md

**Review by SeoFood — APPROVED** [review]
Thanks for the update. I re-tested the current head `cd584a2`.

The previous blocker is fixed: `t=tvsearch&season=01&limit=5` now returns `total="0"` instead of generic results across unrelated shows, while `t=tvsearch&q=Markus%20Lanz&limit=5` still returns usable generic results. I also verified the whitespace-only query case returns no generic results.

The non-blocking title cleanup is fixed as well: titles like `Staffel 2 Folge 3` now produce `S02E03.GERMAN...` instead of duplicating the parsed pattern.

Local verification: lint, format check, typecheck, tests (50 passed), production build, Docker build, and live Docker smoke test on `localhost:6768`.
